runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -1181,6 +1248,43 @@ TR_J9InlinerPolicy::createUnsafePutWithOffset(TR::ResolvedMethodSymbol *calleeSy
    // Preserve null check on the unsafe object
    TR::TransformUtil::separateNullCheck(comp(), callNodeTreeTop, tracer()->debugLevel());
 
+
+   bool conversionNeeded = comp()->fe()->dataTypeForLoadOrStore(type) != type;
+
+   // If we are not able to get javaLangClass it is still inefficient to put direct Access far
+   // So in that case we will generate lowTagCmpTest to branch to indirect access if true
+   TR_OpaqueClassBlock *javaLangClass = comp()->getClassClassPointer(/* isVettedForAOT = */ true);
+   bool needNotLowTagged = javaLangClass != NULL  || conversionNeeded ;
+
+   int length;
+   const char *objTypeSig = unsafeCall->getChild(1)->getSymbolReference() ? unsafeCall->getChild(1)->getSymbolReference()->getTypeSignature(length) : NULL;
+
+   // There are four cases where we cannot be sure of the Object type at compile time:
+   // 1.) Object type info is unknown/inaccessible
+   // 2.) Object type is known at compile time to be java/lang/Object
+   // 3.) Object type is known at compile time to be one of the interface classes java/lang/Cloneable or java/io/Serializable
+   bool objTypeUnknown = objTypeSig == NULL || strncmp(objTypeSig, "Ljava/lang/Object;", strlen("Ljava/lang/Object;")) == 0
```

Comment:
check out abdul's latest work on this: it is JLO (java/lang/Object) or any interface type. change correspondingly.